lib/components/molecules/Pagination/Pagination.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+/**
+ *
+ * Pagination
+ *
+ */
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+import styles from './Pagination.style';
+import type { Node } from './types';
+import { noop } from '@babel/types';
+
+type Props = {
+  size: number,
+  render: Function,
+  items: Array,
+  className: Array | string,
+};
+const buttons = (n, component) => Array.from('x'.repeat(n), (_, s) => component(s + 1));
+const Pagination = ({ size, render, items }: Props): Node => {
+  const [current, setCurrentPage] = useState(1);
+  const pages = Math.ceil(items.length / size);
+  const button = n => (
+    <button key={n} onClick={() => setCurrentPage(n)}>
```

Comment:
This onClick={() => setCurrentPage(n)} forms a closure with-in the component and quite bad for React from performance stand point